lib/ArduinoNative/src/WString.h

```diff
@@ -290,7 +291,27 @@ class String
 
                 delete[] m_buffer;
                 m_buffer = tmp;
-                m_size += str.m_size;
+                m_size += str.m_size - 1;
+            }
+        }
+
+        return *this;
+    }
+
+    String& operator+=(const char* cstr)
+    {
+        if (nullptr != cstr)
+        {
+            char* tmp = new char[m_size + strlen(cstr)];
+
+            if (nullptr != tmp)
+            {
+                strcpy(tmp, m_buffer);
+                strcat(tmp, cstr);

```

Comment:
faster then strcat():
strcpy( & tmp[m_size], cstr);

Reply:
m_size index would be off by 1 because of the null termination!!
A working version would be ```strcpy(&tmp[m_size-1], cstr)```, but then the strcat code looks way cleaner!

Reply:
sure cleaner but ssssllloooowwwwwweeeeeeerrrrrrrr.  strcat() needs to scan to the end of the destination string and then do a strcpy() there. As we know the offset to copy already, we can  save the world some energy.

Reply:
Implemented in [bf66f40](https://github.com/BlueAndi/DroidControlShip/pull/30/commits/bf66f40a8fcd14f0ad79000dcc651935578d2192)